actions/global_actions.jsx

```diff
@@ -291,47 +291,61 @@ export async function redirectUserToDefaultTeam() {
     const locale = getCurrentLocale(state);
     const teamId = LocalStorageStore.getPreviousTeamId(user.id);
 
-    let team = getTeam(state, teamId);
-    const myMember = getMyTeamMember(state, teamId);
+    let channel;
+    let team;
+    let myTeams = getMyTeams(state);
 
-    if (!team || !myMember || !myMember.team_id) {
-        team = null;
-        let myTeams = getMyTeams(state);
-
-        if (myTeams.length > 0) {
-            myTeams = filterAndSortTeamsByDisplayName(myTeams, locale);
-            if (myTeams && myTeams[0]) {
-                team = myTeams[0];
-            }
-        }
+    if (myTeams.length === 0) {
+        browserHistory.push('/select_team');
+        return;
     }
 
-    if (user.id && team) {
-        if (Utils.isGuest(user)) {
-            await dispatch(fetchMyChannelsAndMembers(team.id));
-            state = getState();
+
+    myTeams = filterAndSortTeamsByDisplayName(myTeams, locale);
+
+    for (let myTeam of myTeams) {
+        const myMember = getMyTeamMember(state, myTeam.id);
+        if (!myMember) {
+            continue;
         }
 
-        let channelName = LocalStorageStore.getPreviousChannelName(user.id, team.id);
-        let channel = getChannelByName(state, channelName);
-        const channelMember = getMyChannelMember(state, channelName);
+        if (user.id && myTeam) {
+            if (user && Utils.isGuest(user)) {
```

Comment:
Why are we checking `user.id` first and then `user`? We already assume those are defined here:

https://github.com/mattermost/mattermost-webapp/blob/63bc1640e66fc895d863f4d32c8e3f6955942e61/actions/global_actions.jsx#L292

And if it's possible for `user` to be `undefined` we should return before that statement.

Also we assume `myTeam` to be defined here:

https://github.com/mattermost/mattermost-webapp/blob/63bc1640e66fc895d863f4d32c8e3f6955942e61/actions/global_actions.jsx#L307

Reply:
Thanks, I realized now that if you don't have user, makes no sense to keep going, so I'll ignore it, and all the `myTeams` members should be a valid object, so we can remove the check.